be/src/vec/common/hash_table/hash_table_key_holder.h

```diff
@@ -120,7 +120,7 @@ namespace doris::vectorized {
   */
 struct SerializedKeyHolder {
     StringRef key;
-    Arena& pool;
+    Arena* pool;
```

Comment:
why do you change this and I find you change a lot of related code.

Reply:
To avoid calling `get_key_holder` for each row twice, `KeyHolder` should be cached in a vector, but `SerializedKeyHolder` contains reference of Arena cannot be copied into the vector.